classes/migration/upgrade/v3_4_0/I7014_DoiMigration.inc.php

```diff
@@ -250,13 +250,18 @@ private function _migrateDoiSettingsToContext()
                 $carry[] = [
                     'journal_id' => $item->journal_id,
                     'setting_name' => 'enableDois',
-                    'setting_value' => 0,
+                    'setting_value' => 1,
                 ];
                 $carry[] = [
                     'journal_id' => $item->journal_id,
                     'setting_name' => 'doiCreationTime',
                     'setting_value' => 'copyEditCreationTime'
                 ];
+                $carry[] = [
+                    'journal_id' => $item->journal_id,
+                    'setting_name' => 'useDefaultDoiSuffix',
+                    'setting_value' => 1
+                ];
```

Comment:
I _think_, if the DOI plugin was not previously enabled, then the journal has opted out of them. In that case, I don't think we want to enable DOIs on upgrade. In such cases, the journal should not enable them. 

It's only on a fresh install that DOIs should be enabled.